### DISCUSSION TITLE
快照版 2.7.4 更新日志

### DISCUSSION THREAD

**Discussion by LTCatt** [Meloong-Git/PCL#3895] [讨论]
· MC 安装优化
　　会自动隐藏明显不可用的自动安装选项（#3841）
　　在 1.16+ 同时安装 OptiFine 和 Mod 加载器时显示兼容性提示（MCLF-CN/docs#6）
· 正版登录优化
　　暂时强制改用浏览器登录新账号（#3849，@kong-que-dnf123）
　　或许避免了需要先用官启登录一次的情况（#3702，@zkitefly，by @allMagicNB）
　　添加部分登录失败的处理建议（#3730，@JingHai-Lingyun）
　　添加登录 SSL 证书验证失败的提示（#3846，@xxtg666）
· 崩溃分析优化
　　游戏崩溃时会在日志中输出系统诊断信息
　　添加了 OptiFine 1.20 与 Forge 不兼容的崩溃分析
　　支持分析堆栈信息中的 mixin 注入方法
　　为 Mod 导致的崩溃提供更明确的解决方案描述
　　添加了其他 5 种崩溃的分析
· 自定义主页：允许为 MyListItem 设置联网帮助页面（#3266，@Xiaolv114518）
· 支持从版本 jar 中获取版本号（#3294，@Zhichii）
· 在 64 位系统上彻底禁用 32 位 Java（#3649，@zhangyankang521）
· 添加部分披风的中文翻译（#3840，#3850，#3883，@Light-Beacon，by @xiaoliyuanpp，by @Deep-Dark-Forest）
· 在无法打开纯文本文件时给出提示（#3884，@Leonmmcoset）
· 缓解 MC 的突发掉帧现象，将 -Xmn 从 50% 调整至 15%（#3891，@Hlwyefeng）
· 新主页预设：Minecraft 皮肤推荐（by @wkea）
· 使用 Queue 作为小文件下载缓存，以优化性能
· 不再将空文件夹列入版本列表
· 修复了以下 20 个 Bug
　　特定情况下无法关闭失败的下载任务卡片（#3007，@UserNameiiiii，@kong-que-dnf123）
　　删除选中的正版账号后未重置下拉框（#3198，@MGWtino0）
　　同一 Mod 在不同 MC 版本下使用了相同的更新检查缓存（#3680，@HMwancheng）
　　自定义主页：导入整合包事件无效（#3695，@wkea）
　　在结束后开始音乐播放有可能出错（#3708，@wuwenjun9939）
　　极端情况下联网下载自定义主页出错（#3721，@mc-124）
　　部分 Modrinth 支持库被错误地识别（#3725，@Dave12138）
　　从 MC 更新提示点击下载可能导致出错（#3742，@U3L6）
　　极低概率下获取任务栏进度可能出错（#3771，@Casper233）
　　极低概率下加载 MC 版本列表可能失败（#3791，@wuliaodexiaoluo）
　　同时安装多个部件可能导致尝试下载多次 Java（#3797，@wuliaodexiaoluo）
　　整合包安装过程中会被标记为错误的版本（#3809，@JingHai-Lingyun）
　　删除 Mod 时可能出错（#3816，@GCatMew000）
　　新安装的整合包可能不会将语言自动设置为中文（#3844，@candybits）
　　微软账户登录页面无法加载（#3849，@kong-que-dnf123）
　　优先选择 JDK 23 而非 JDK 21 启动游戏（#3851，@xtrackc1437）
　　会尝试自动下载 Java 7（#3873）
　　下载 Mod 时的提示可能显示 Mod 版本号而非文件名
　　手动拖入崩溃报告文件进行分析时的分析结果描述有误
　　无法正确分析堆栈溢出时的崩溃报告

<hr /><em>This discussion was created from the release <a href='https://github.com/Hex-Dragon/PCL2/releases/tag/2.7.4'>2.7.4</a>.</em>

**Comment by allMagicNB** [comment]
@LTCatt #3702 不是我交的 PR 修的吗，咋没我名字（

**Reply by LTCatt** [reply]
草，by 标错了

**Comment by Deep-Dark-Forest** [comment]
@LTCatt 这里是不是少个#3850（
![Screenshot_20240525_203932.jpg](https://github.com/Hex-Dragon/PCL2/assets/131975043/27f7738d-d3ac-458d-80c9-7ad26a40b14d)

**Reply by LTCatt** [reply]
补了
